presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/TestMergeAdjacentWindows.java

```diff
@@ -114,8 +115,49 @@ public void testDoesNotFireForNonWindowIntermediateNode()
                                                 newWindowNodeSpecification(p, "a"),
                                                 ImmutableMap.of(p.symbol("avg_1", BIGINT), newWindowNodeFunction("avg", "a")),
                                                 p.values(p.symbol("a", BIGINT), p.symbol("b", BIGINT))
+                                        )
                                 )
-                        )))
+                        ))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testDependentAdjacentWindowsIdenticalSpecifications()
+            throws Exception
+    {
+        Optional<Window> windowA = Optional.of(new Window(ImmutableList.of(new SymbolReference("a")), ImmutableList.of(), Optional.empty()));
+
+        tester.assertThat(new MergeAdjacentWindows())
+                .on(p ->
+                        p.window(
+                                newWindowNodeSpecification(p, "a"),
```

Comment:
can you please extract specification as variable, then it will be clear that both window nodes are using the same (mergable) specification